### ISSUE TITLE
Behaviour for hooks for RewardOverridenEnv

### ISSUE THREAD

**Issue by Sid-Bhatia-0** [JuliaReinforcementLearning/ReinforcementLearningCore.jl#161]
I am trying to use `RewardOverridenEnv` for a per-step penalty to be added in addition to the regular reward of the underlying environment.

The reward related hooks, like `TotalRewardPerEpisode`, for example, explicitly override the behavior to record the original reward (without penalty). This means that while logging reward information (using `TensorBoardLogger`, say), I am unable to view the intended reward, which feels a little unintuitive and misleading, because it is ultimately the final reward after all the transformations that I care about, which is why I used the `RewardOverridenEnv` wrapper in the first place.

**Comment by findmyway** [comment]
> for example, explicitly override the behavior to record the original reward (without penalty).

Yes, that's intentional for Atari related experiments. But a better solution is to make it configurable.
